pkg/cloud/oracle/provider.go

```diff
@@ -0,0 +1,314 @@
+	v1 "k8s.io/api/core/v1"
+)
+
+const nodePoolIdAnnotation = "oci.oraclecloud.com/node-pool-id"
+const virtualPoolIdAnnotation = "oci.oraclecloud.com/virtual-node-pool-id"
+const virtualNodeLabel = "node-role.kubernetes.io/virtual-node"
+const managementPlatformOKE = "oke"
+const currencyCodeUSD = "USD"
+
+type Oracle struct {
+	Config                  models.ProviderConfig
+	Clientset               clustercache.ClusterCache
+	ClusterRegion           string
+	ClusterAccountID        string
+	DownloadPricingDataLock sync.RWMutex
+	OSEnvLock               sync.Mutex
+	RateCardStore           *RateCardStore
+	ServiceAccountChecks    *models.ServiceAccountChecks
+	DefaultPricing          DefaultPricing
+}
+
+func (o *Oracle) ClusterInfo() (map[string]string, error) {
+	c, err := o.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	m := make(map[string]string)
+	m["name"] = "Oracle Cluster #1"
+	if clusterName := o.getClusterName(c); clusterName != "" {
+		m["name"] = clusterName
+	}
+	m["provider"] = opencost.OracleProvider
+	m["account"] = o.ClusterAccountID
+	m["region"] = o.ClusterRegion
+	m["remoteReadEnabled"] = strconv.FormatBool(env.IsRemoteEnabled())
+	m["id"] = env.GetClusterID()
+	return m, nil
+}
+
+func (o *Oracle) NodePricing(key models.Key) (*models.Node, models.PricingMetadata, error) {
+	if err := o.ensurePricingData(); err != nil {
+		return nil, models.PricingMetadata{}, err
+	}
+	o.DownloadPricingDataLock.RLock()
+	defer o.DownloadPricingDataLock.RUnlock()
+	return o.RateCardStore.ForKey(key, o.DefaultPricing)
+}
+
+func (o *Oracle) PVPricing(pvk models.PVKey) (*models.PV, error) {
+	if err := o.ensurePricingData(); err != nil {
+		return nil, err
+	}
+	o.DownloadPricingDataLock.RLock()
+	defer o.DownloadPricingDataLock.RUnlock()
+	return o.RateCardStore.ForPVK(pvk, o.DefaultPricing)
+}
+
+func (o *Oracle) NetworkPricing() (*models.Network, error) {
+	if err := o.ensurePricingData(); err != nil {
+		return nil, err
+	}
+	o.DownloadPricingDataLock.RLock()
+	defer o.DownloadPricingDataLock.RUnlock()
+	return o.RateCardStore.ForEgressRegion(o.ClusterRegion, o.DefaultPricing)
+}
+
+func (o *Oracle) LoadBalancerPricing() (*models.LoadBalancer, error) {
+	if err := o.ensurePricingData(); err != nil {
+		return nil, err
+	}
+	o.DownloadPricingDataLock.RLock()
+	defer o.DownloadPricingDataLock.RUnlock()
+	return o.RateCardStore.ForLB(o.DefaultPricing)
+}
+
+func (o *Oracle) AllNodePricing() (interface{}, error) {
+	if err := o.ensurePricingData(); err != nil {
+		return nil, err
+	}
+	o.DownloadPricingDataLock.RLock()
+	defer o.DownloadPricingDataLock.RUnlock()
+	return o.RateCardStore.Store(), nil
+}
+
+// DownloadPricingData refreshes the RateCardStore pricing data.
+func (o *Oracle) DownloadPricingData() error {
+	o.DownloadPricingDataLock.Lock()
+	defer o.DownloadPricingDataLock.Unlock()
+	cfg, err := o.GetConfig()
+	if err != nil {
+		return err
+	}
+	if o.RateCardStore == nil {
+		url := os.Getenv("OCI_PRICING_URL")
```

Comment:
Could you instead do this in https://github.com/opencost/opencost/blob/develop/pkg/env/costmodelenv.go and supply a default for pricing data that would work for most users?